registry/storage/catalog.go

```diff
@@ -49,17 +48,22 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 		// if we've filled our array, no need to walk any further
 		if len(foundRepos) == len(repos) {
```

Comment:
This is done here for safety. We really don't want this check getting mucked up in the logic above, if it changes.